app/services/encrypted_attribute.rb

```diff
@@ -0,0 +1,76 @@
+class EncryptedAttribute
+  attr_accessor :user_access_key
+  attr_reader :encrypted, :decrypted
+
+  def self.new_user_access_key(cost: nil, key: nil)
+    env = Figaro.env
+    key ||= env.attribute_encryption_key
+    cost ||= env.attribute_cost
+    UserAccessKey.new(password: key, salt: key, cost: cost)
+  end
+
+  def self.new_from_decrypted(decrypted, user_access_key = new_user_access_key)
+    encryptor = Pii::PasswordEncryptor.new
+    encrypted = encryptor.encrypt(decrypted, user_access_key)
+    new(encrypted, decrypted: decrypted, user_access_key: user_access_key)
+  end
+
+  def initialize(encrypted, decrypted: nil, cost: nil, user_access_key: nil)
+    self.encrypted = encrypted
```

Comment:
Why not use an ivar here instead? I had to go into irb to make sure I understood what was happening here... oh `self`... 

For those watching at home:

```
class Hello
  attr_reader :the_method

  def initialize(arg: 'hello')
    self.the_method = arg
  end

  private

  attr_writer :the_method
end

Hello.new.the_method
=> "hello"
```

I usually see it written like this:

```
class Hello
  attr_reader :the_method

  def initialize(arg: 'hello')
   @the_method = arg
  end
end

Hello.new.the_method
=> "hello"
```

What do you think? I do not see a reason while the latter would not work, but I could be missing something.